irods/client_init.py

```diff
@@ -53,20 +54,52 @@ def write_native_credentials_to_secrets_file(password, overwrite=True, **kw):
     _write_encoded_auth_value(auth_file, password, overwrite)
 
 
-def write_pam_credentials_to_secrets_file(password, overwrite=True, **kw):
+# Reverse compatibility.
+write_native_credentials_to_secrets_file = write_native_irodsA_file
+
+
+# Facility for writing authentication secrets file.  Designed to be useable for iRODS 4.3(+) and 4.2(-).
+def write_pam_irodsA_file(password, overwrite=True, ttl = '', **kw):
+    import irods.auth.pam_password
+    from irods.session import iRODSSession
+    import io
+    ses = kw.pop('_session', None) or h.make_session(**kw)
+    if ses._server_version(iRODSSession.RAW_SERVER_VERSION) < (4,3):
+        write_pam_credentials_to_secrets_file(
+            password,
+            overwrite = overwrite,
+            ttl = ttl,
+            _session = ses)
```

Comment:
What is this for?
Is there some difference between 4.2 and 4.3 regarding the .irodsA file?

Reply:
No, except that `write_pam_irodsA_file` is intentioned to work whether the server pointed at is 4.2- or 4.3+

`write_pam_credentials_to_secrets_file` will only work for legacy auth, i.e. until we remove that feature from some server version >` 4.3.3`

Reply:
That answers my question.

Will need to document that in the README so users aren't confused about which one to use moving forward.

Reply:
> Should we `return` early in this case? It appears to continue to the logic below as it is.

Good catch.

Reply:
Done.  And thanks for bringing the question to light, @korydraughn .